docs/interfaces/auth-service.txt

```diff
@@ -0,0 +1,45 @@
+Interface Token {
+    payload: {
+        // The lifetime of the Token in milliseconds.
+        expiresIn: number (ms),
+        // The corresponding userId of the requesting client.
+        userId: string,
+        // The id of the current session.
+        sessionId: string
+    },
+    signature: string
+}
+
+/**
+* The client can login with its credentials for authentication.
+* If they are correct, the service answers with a signed Token and sets a cookie, containing the sessionId of the client. 
+*/
+(Token, Cookie) Login (username: string, password: string);
+
+/**
+* An example for any protected route. If the client requests protected resources, it has to 
+* send the signed Token and the cookie, it receives from the service at login, to the server.
+*/
+Any Api* (token: Token, cookie: Cookie);
```

Comment:
Suggestion: Name this `authenticate` and define exatly, what is returned, e.g. `{userId: number; sessionId: string;}`. Please make clear, that this is a library function.